src/riak_core_info_service_process.erl

```diff
@@ -0,0 +1,69 @@
+-module(riak_core_info_service_process).
+-behaviour(gen_server).
+
+-record(state, {
+    source = undefined :: riak_core_info_service:callback(),
+    response_handler = undefined :: riak_core_info_service:callback(),
+    shutdown = undefined :: riak_core_info_service:callback()
+}).
+
+-export([start_link/4,
+         init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+
+%% `gen_server' implementation
+
+start_link(Registration, Shutdown, Source, ResponseHandler) ->
+    {ok, Pid} = gen_server:start_link(?MODULE, [Shutdown, Source, ResponseHandler], []),
```

Comment:
Trivial, but why not just create the `#state{}` record here and pass it through intact?